x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/OperatorTestCase.java

```diff
@@ -80,15 +83,26 @@ public final void testSimpleLargeInput() {
      * in a sane way.
      */
     public final void testSimpleCircuitBreaking() {
-        BigArrays bigArrays = new MockBigArrays(PageCacheRecycler.NON_RECYCLING_INSTANCE, smallEnoughToCircuitBreak());
-        CircuitBreaker breaker = bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST);
-        BlockFactory blockFactory = BlockFactory.getInstance(breaker, bigArrays);
-        Exception e = expectThrows(
-            CircuitBreakingException.class,
-            () -> assertSimple(new DriverContext(bigArrays, blockFactory), between(1_000, 10_000))
-        );
-        assertThat(e.getMessage(), equalTo(MockBigArrays.ERROR_MESSAGE));
-        assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L));
+        /*
+         * We build two CircuitBreakers - one for the input blocks and one for the operation itself.
+         * The input blocks don't count against the memory usage for the limited operator that we
+         * build.
+         */
+        List<Page> input = CannedSourceOperator.collectPages(simpleInput(driverContext().blockFactory(), between(1_000, 10_000)));
+        try {
+            BigArrays bigArrays = new MockBigArrays(PageCacheRecycler.NON_RECYCLING_INSTANCE, smallEnoughToCircuitBreak())
+                .withCircuitBreaking();
+            CircuitBreaker breaker = bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST);
+            BlockFactory blockFactory = BlockFactory.getInstance(breaker, bigArrays);
+            Exception e = expectThrows(
+                CircuitBreakingException.class,
+                () -> drive(simple(bigArrays).get(new DriverContext(bigArrays, blockFactory)), input.iterator())
+            );
+            assertThat(e.getMessage(), equalTo(MockBigArrays.ERROR_MESSAGE));
+            assertThat(bigArrays.breakerService().getBreaker(CircuitBreaker.REQUEST).getUsed(), equalTo(0L));
+        } finally {
+            Releasables.close(() -> Iterators.map(input.iterator(), p -> () -> p.releaseBlocks()));
```

Comment:
Boom! my head exploded!! ;-) So this is inferred to `Iterators.<Page, Releasable>map(..`. Maybe just use a method reference here to avoid the extra lambda arrows ( as suggested by your favourite IDE! ) 
```
Iterators.map(input.iterator(), p -> p::releaseBlocks)
```

Reply:
That kept not working. Let me try again.